streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollector.java

```diff
@@ -45,8 +44,6 @@
                      final Serializer<V> valueSerializer,
                      final StreamPartitioner<? super K, ? super V> partitioner);
 
-    void commit(final Map<TopicPartition, OffsetAndMetadata> offsets);
```

Comment:
We moved this to `TaskManager`

Reply:
Please see my other comment above --- I think it is cleaner to just call `foreach(active-task) task.recordCollector.commit` inside the task-manager; and inside RecordCollectorImpl we check that eosEnabled is always true, otherwise illegal-state thrown.

In the next PR where we have the thread-producer, we could then only create a single `recordCollector` object that is shared among all active tasks and wraps the thread-producer, and then the caller `taskManager` code then can just get one active task and call its record-collector's commit function knowing that is sufficient to actually commit for all tasks since everyone is using the same record-collector.

WDYT?